tests/Utils/TempDirectoryCreatorTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Utils;
+
+use Infection\Filesystem\Filesystem;
+use Infection\Utils\TempDirectoryCreator;
+use PHPUnit\Framework\TestCase;
+
+class TempDirectoryCreatorTest extends TestCase
+{
+    /**
+     * @var TempDirectoryCreator
+     */
+    private $creator;
+
+    /**
+     * @var string
+     */
+    private $tempDir;
+
+    protected function setUp()
+    {
+        $this->creator = new TempDirectoryCreator(new Filesystem());
+        $this->tempDir = sys_get_temp_dir();
+    }
+
+    protected function tearDown()
+    {
+        @unlink($this->tempDir);
```

Comment:
if we have the fileSystem instance here we might as well use its `remove()` function?

Reply:
we don't have this method in our version of the Filesytem yet and I don't see any sense to create it just for tests.

Reply:
you could just use the Symfony one: 

```php
(new SymfonyFilesystem())->remove($this->tmpDir);
```

At least we would be sure the content of that directory is recursively removed as right now I'm not even sure this is working

Reply:
we don't have `symfony/filesystem` as dependency but if you insist I'll backport `remove` method as well

Reply:
Then I would do so yes. You can easily double check that it's not working as it is:

```
$ mkdir tmp
$ touch tmp/foo

$ psysh
>>> unlink('tmp')
PHP Warning:  unlink(tmp): Operation not permitted on line 1
```

So we are actually not cleaning up at all here